src/proto/generation.proto

```diff
@@ -97,25 +94,53 @@ message SamplerParameters {
   optional float cfg_scale = 5;
 }
 
+// Unused, but reserved for future use. Adjustments to the latents after
+// initialization.
 message ConditionerParameters {
   optional string vector_adjust_prior = 1;
 }
 
+// Future, unimplemented.
 enum Upscaler {
   UPSCALER_RGB = 0;
   UPSCALER_GFPGAN = 1;
   UPSCALER_ESRGAN = 2;
 }
 
+// When does this schedule definition apply?
 message ScheduleParameters {
   optional float start = 1;     // 0.0 to 1.0
   optional float end = 2;       // 0.0 to 1.0
 }
 
+// Parameters that apply to this block of the schedule.
 message StepParameter {
   float scaled_step = 1;
   optional SamplerParameters sampler = 2;
   optional ScheduleParameters schedule = 3;
+  optional GuidanceParameters guidance = 4;
+}
+
+// What type of guidance to use.
+enum GuidanceType {
+  GUIDANCE_CFG = 0;               // default is CFG guidance
+  GUIDANCE_CLIP = 1;
```

Comment:
well this is CFG+CLIP actually... a better naming?

Reply:
Changed to `GUIDANCE_CFG_CLIP`.